pkg/bgpv1/manager/reconcilerv2/neighbor.go

```diff
@@ -148,12 +188,27 @@ func (r *NeighborReconciler) Reconcile(ctx context.Context, p ReconcileParams) e
 		}
 
 		if n.PeerAddress == nil {
-			r.logger.Debug("Peer does not have PeerAddress configured, skipping", types.PeerLogField, n.Name)
-			continue
+			// future auto-discovery modes can be added here to get the peer address
+			switch n.AutoDiscovery.Mode {
+			case v2.BGPDefaultGatewayMode:
+				defaultGateway, err := r.getDefaultGateway(n.AutoDiscovery.DefaultGateway)
+				if err != nil {
+					r.logger.Debug("failed to get default gateway, skipping",
+						types.PeerLogField,
+						n.Name,
+						logfields.Error,
+						err)
+					continue
+				}
+				newNeigh[i].PeerAddress = &defaultGateway
+			default:
+				r.logger.Debug("Peer does not have PeerAddress configured, skipping", types.PeerLogField, n.Name)
+				continue
+			}
 		}
 
 		var (
-			key = r.neighborID(&n)
+			key = r.neighborID(&newNeigh[i])
```

Comment:
Why do we need this change?

Reply:
because we are updating h.new i.e new peers with `newNeigh[i]` instead of `n` in the following line so i wanted to keep it consistent with that instead of updating that part of the code to use `n`. i'm open to both the options. 
https://github.com/cilium/cilium/pull/39416/files#diff-c06abd174799f2fedd8fc17f5fcfe231ee52a690a45c9d5d4b2541774260f012L185